ecs-init/cache/cache.go

```diff
@@ -76,11 +99,38 @@ func NewDownloader() (*Downloader, error) {
 	return downloader, nil
 }
 
+// AgentCacheStatus inspects the on-disk cache and returns its
+// status. See `CacheStatus` for possible cache statuses and
+// scenarios.
+func (d *Downloader) AgentCacheStatus() CacheStatus {
+	stateFile := config.CacheState()
+	// State file and tarball must be non-zero to report status on
+	uncached := !(d.fileNotEmpty(stateFile) && d.fileNotEmpty(config.AgentTarball()))
+	if uncached {
+		return StatusUncached
+	}
+
+	file, err := d.fs.Open(stateFile)
+	if err != nil {
+		return StatusUncached
+	}
+	var status CacheStatus
+	_, err = fmt.Fscanf(file, "%d", &status)
+	if err != nil {
+		return StatusUncached
+	}
+	return status
```

Comment:
It looks like you could have an invalid `CacheStatus` like `3` be returned here.  It would be better to force that to `StatusUncached` than ignore it.

Reply:
Yes, you can have an invalid status returned here. This was intentionally left unhandled.